### ISSUE TITLE
New `godoclint` rule and potential breaks with next Golangci-lint release

### ISSUE THREAD

**Issue by babakks** [mgechev/revive#1617]
*This is a heads up before the upcoming Golangci-lint release.*

The new version of `godoclint` (See [`v0.11.1`](https://github.com/godoc-lint/godoc-lint/releases/tag/v0.11.1)) which will be shipped with the next Golangci-lint release, [introduces a new rule named `require-stdlib-doclink`](https://github.com/godoc-lint/godoc-lint?tab=readme-ov-file#require-stdlib-doclink) to enforce using proper doc links when referencing stdlib symbols.

Since all `godoclint` rules are enabled in `.golangci.yml`, upgrading Golangci-lint will result in new issues/errors (expand for details). The fix is basically surrounding the suggested symbols with square brackets to make them doc links. Needless to say you can also disable this new rule by adding `require-stdlib-doclink` to `linters.settings.godoclint.disable` in `.golangci.yml`, but IMHO keeping it helps with much nicer godocs.

Also, `godoclint` will no longer complain about require godoc for exported methods on unexported types. I noticed the following exclusion in `.golangci.yml`, which will be redundant after upgrade (also see the first line in the logs below):

```yaml
linters:
  exclusions:
    rules:
      - linters:
          - godoclint
        text: 'symbol should have a godoc \("Visit"\)'
```

*If you're up to for the fix, I'd be happy to contribute back to Revive.* 🍻

### Issues 
<details><summary>Details</summary>
<p>

```
...
WARN [runner/exclusion_rules] Skipped 0 issues by rules: [Text: "symbol should have a godoc \\(\"Visit\"\\)", Linters: "godoclint"] 
...
internal/ifelse/branch.go:17:1: text "ast.BlockStmt" should be replaced with "[ast.BlockStmt]" to link to stdlib type (godoclint)
// BlockBranch gets the Branch of an ast.BlockStmt.
^
internal/ifelse/branch.go:29:1: text "ast.Stmt" should be replaced with "[ast.Stmt]" to link to stdlib type (godoclint)
// StmtBranch gets the Branch of an ast.Stmt.
^
lint/package.go:145:1: text "go/types.Config.Check" should be replaced with "[go/types.Config.Check]" to link to stdlib method (godoclint)
// check function encapsulates the call to go/types.Config.Check method and
^
rule/cognitive_complexity.go:98:1: text "ast.Visitor" should be replaced with "[ast.Visitor]" to link to stdlib type (godoclint)
// Visit implements the ast.Visitor interface.
^
rule/cyclomatic.go:102:1: text "ast.Visitor" should be replaced with "[ast.Visitor]" to link to stdlib type (godoclint)
// Visit implements the ast.Visitor interface.
^
rule/package_directory_mismatch.go:79:1: text "filepath.Base" should be replaced with "[filepath.Base]" to link to stdlib function (godoclint)
// skipDirs contains directory names that should be unconditionally ignored when checking.
^
rule/time_date.go:17:1: text "time.Date" should be replaced with "[time.Date]" to link to stdlib function (godoclint)
// TimeDateRule lints the way time.Date is used.
^
rule/time_date.go:44:1: text "time.Date" should be replaced with "[time.Date]" to link to stdlib function (godoclint)
// timeDateArgument is a type for the arguments of time.Date function.
^
rule/time_date.go:59:2: text "time.Date" should be replaced with "[time.Date]" to link to stdlib function (godoclint)
	// timeDateArgumentNames are the names of the arguments of time.Date.
	^
rule/time_date.go:71:2: text "time.Date" should be replaced with "[time.Date]" to link to stdlib function (godoclint)
	// timeDateArity is the number of arguments of time.Date.
	^
rule/time_equal.go:12:1: text "time.Time" should be replaced with "[time.Time]" to link to stdlib type (godoclint)
// TimeEqualRule shows where "==" and "!=" used for equality check time.Time.
^
rule/use_errors_new.go:10:1: text "fmt.Errorf" should be replaced with "[fmt.Errorf]" to link to stdlib function (godoclint)
// UseErrorsNewRule spots calls to fmt.Errorf that can be replaced by errors.New.
^
rule/use_errors_new.go:10:1: text "errors.New" should be replaced with "[errors.New]" to link to stdlib function (godoclint)
// UseErrorsNewRule spots calls to fmt.Errorf that can be replaced by errors.New.
^
rule/var_naming.go:439:1: text "unicode.IsDigit" should be replaced with "[unicode.IsDigit]" to link to stdlib function (godoclint)
// isDigit checks if rune is a simple digit.
^
rule/var_naming.go:446:1: text "unicode.IsUpper" should be replaced with "[unicode.IsUpper]" to link to stdlib function (godoclint)
// isUpper checks if rune is ASCII upper case letter
^
rule/waitgroup_by_value.go:10:1: text "sync.WaitGroup" should be replaced with "[sync.WaitGroup]" to link to stdlib type (godoclint)
// WaitGroupByValueRule lints sync.WaitGroup passed by copy in functions.
^
16 issues:
* godoclint: 16
```

</p>
</details> 

cc @ccoVeille @alexandear

**Comment by alexandear** [comment]
@babakks thank you for the issue and for offering to help.
I've fixed this myself already - it wasn't hard. Appreciate you flagging it in advance.
